realsense/plugin.cpp

```diff
@@ -39,7 +38,10 @@ class realsense : public plugin {
         // Even if the API does not invoke `callback` in parallel, this is still important for the memory-model.
         // Without this lock, prior invocations of `callback` are not necessarily "happens-before" ordered, so accessing
         // persistent variables constitutes a data-race, which is undefined behavior in the C++ memory model.
-
+        if (!_m_clock->has_started()) {
+            // if the data arrived before the relative clock started, ignore
+            return;
+        }
```

Comment:
Now that you've moved `configure_camera()` to `start()`, the clock is guaranteed to be ready before the first callback occurs. Or am I missing something?

Reply:
Relative clcok is currently started after all plugins are [started](https://github.com/ILLIXR/ILLIXR/blob/024494818bd7834a0c70d08d58bd2746ffc512a7/runtime/runtime_impl.hpp#L64-L68). So moving that `configure_camera()` to `start()` won't fix the problem as the first callback might already happen before the clock starts. Instead, in all plugins inherited from threadloop, we are using [`Stoplight`](https://github.com/ILLIXR/ILLIXR/blob/024494818bd7834a0c70d08d58bd2746ffc512a7/common/threadloop.hpp#L89) to guarantee the start of Relative Clock, but nothing like that exists for non-threadloop plugins like RealSense. We can add a "if started" checking to Relative Clock for future use cases, which I doubt if there would be many (perhaps only for sensor plugins like RealSense that is neither evoked by any other plugins nor inherited from threadloop).

Reply:
My bad. I was thinking of `threadloop->setup()` when I wrote the above. I see your point. The above solution will definitely work, but I'm not sure it's the cleanest way from a design perspective. In some ways, we've seen this issue before, and I'm wondering if this is a good time to revisit [that discussion](https://github.com/ILLIXR/ILLIXR/pull/250#discussion_r858174782).

Reply:
I think two stoplight events is the best solution.